tracetree.h

```diff
@@ -76,7 +76,7 @@ class TraceTree
             if (it == parent->children.end() || it->instructionPointer != ip) {
                 index = m_index++;
                 it = parent->children.insert(it, {ip, index, {}});
-                fprintf(out, "t %" PRIxPTR " %x\n", reinterpret_cast<uintptr_t>(ip), parent->index);
```

Comment:
Now I understand what you are saying, but the "accepted answer" on StackOverflow is not always the correct answer -- and also, sometimes the correct answer may vary depending on the particular requirements. If you want maximum portability, the correct answer is actually slightly below (quoted at bottom here). I should mention that at the time I was probably trying something which I later realized was not part of your scope (compiling on a 64-bit OS). I simply saved this change for some other team in my company to pick up, but then our investigation moved into a different direction. I am currently a bit busy but if it helps I can try to build heaptrack again to see what the exact problem is.
>>>

The only portable way to print pointer values is to use the "%p" specifier, which produces implementation-defined output. (Converting to uintptr_t and using PRIxPTR is likely to work, but (a)  uintptr_t was introduced in C99, so some compilers may not support it, and (b) it's not guaranteed to exist even in C99 (there might not be an integer type big enough to hold all possible pointer values.

So use "%p" with sprintf() (or snprintf(), if you have it) to print to a string, and then print the string padding with leading blanks.